### DISCUSSION TITLE
Does it make sense to write/add a formulae for an old version of a software project?

### DISCUSSION THREAD

**Discussion by jace0x21** [Homebrew/discussions#5335] [Writing Formulae/Casks] [answered]
### Output of `brew config`

```shell
n/a
```

### Output of `brew doctor`

```shell
n/a
```

### Description of issue

I noticed that a lot of formulae only support the most recent version of a package. I also noticed some cases where formulae exist for an old version but are keg-only (presumably to avoid issues with other software installs). I think it is sometimes convenient to have a formulae for an old version even if it is keg-only to allow easy fetching of dependencies that are required for the build of a project that depends on an older version of a package.

So my two questions are:
1. Is it discouraged to add formulae for older versions of packages. As a concrete example, there is a formulae for capstone 5.0.1, would it be a bad idea to also add a formulae for capstone 4.0.2?
2. Are there any guides for adding new formulae to brew?

**Comment by carlocab** [comment, marked as answer]
This is answered in https://docs.brew.sh.

See:
https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
https://docs.brew.sh/Acceptable-Formulae
https://docs.brew.sh/Versions

**Comment by jace0x21** [comment]
Got it. That pretty much clears up everything. My apologies for not taking a look there first. Thank you!
